meson.build

```
@@ -33,6 +35,20 @@ add_project_link_arguments(
     language: 'c',
 )
 
+# in release mode, we automatically turn off cast checks and g_assert
+if get_option('buildtype').startswith('release')
```

Perhaps we should follow GLib's convention here?
```suggestion
if get_option('optimization') in ['2', '3', 's']
```
https://github.com/GNOME/glib/blob/b3311e82ee04c90136ff231358d356e8e587f4da/meson.build#L279-L282

That would also turn off cast checks and make `g_assert` no-op when building with `--buildtype=debugoptimized` (not sure if that's desired) or `--buildtype=minsize`, see:
https://mesonbuild.com/Builtin-options.html#build-type-options

---

We could also do this:
```diff
--- a/meson.build
+++ b/meson.build
@@ -35,18 +35,16 @@ add_project_link_arguments(
     language: 'c',
 )
 
-# in release mode, we automatically turn off cast checks and g_assert
-if get_option('buildtype').startswith('release')
-    add_project_arguments('-DG_DISABLE_CAST_CHECKS', language : ['cpp', 'c'])
-    add_project_arguments('-DG_DISABLE_CHECKS', language : ['cpp', 'c'])
-    add_project_arguments('-DG_DISABLE_ASSERT', language : ['cpp', 'c'])
-endif
-
 # in debug mode mode, we automatically enable leak checks and fatal warnings
 # also true for 'debugoptimized'
 if get_option('debug')
     add_project_arguments('-DDEBUG_FATAL', language : ['cpp', 'c'])
     add_project_arguments('-DDEBUG_LEAK', language : ['cpp', 'c'])
+elif get_option('optimization') in ['2', '3', 's']
+    # in release mode, we automatically turn off cast checks and g_assert
+    add_project_arguments('-DG_DISABLE_CAST_CHECKS', language : ['cpp', 'c'])
+    add_project_arguments('-DG_DISABLE_CHECKS', language : ['cpp', 'c'])
+    add_project_arguments('-DG_DISABLE_ASSERT', language : ['cpp', 'c'])
 endif
 
 cc = meson.get_compiler('c')
```
to ensure that building with `--buildtype=debugoptimized` or `--buildtype=minsize` would still enable debug facility.

---

Ah good idea, I'll update.